docs/experiments/generic-pattern-matching.md

```diff
@@ -49,7 +88,7 @@ Generic pattern matching has the following properties:
 
 ## Caveats and limitations
 
-Generic mode should work fine with any human-readable text, as long as it’s primarily based on ASCII symbols. In practice, it might work great with some languages and less well with others. In general, it’s possible or even easy to write code in weird ways that will prevent generic mode from matching. Note it’s not good for detecting malicious code. For example, in HTML one can write `&#x48;&#x65;&#x6C;&#x6C;&#x6F`; instead of `Hello` and this is not something the generic mode would match if the pattern is `Hello`, unlike if it had full HTML support.
+Generic mode should work fine with any human-readable text, as long as it’s primarily based on ASCII symbols. In practice, it might work great with some languages and less well with others. In general, it’s possible or even easy to write code in weird ways that will prevent generic mode from matching. Note it’s not good for detecting malicious code. For example, in XML one can write `&#x48;&#x65;&#x6C;&#x6C;&#x6F`; instead of `Hello` and this is not something the generic mode would match if the pattern is `Hello`, unlike if it had full XML support.
```

Comment:
@adamkvitek I know we're not supposed to use "should" or "might" - but this is a genuinely experimental feature, and we genuinely don't know how it will behave for all possible combinations. I think I'd leave this in because it accurately tells users about the actual state of the product and doesn't mislead them with absolute guarantees?

Reply:
@s-santillan than express the uncertainty in other ways. Technical communication is closer to a sign language on roads sometimes than to a real language (similarly to how we mark optional steps as `optional` in our texts, you can just straight say that this is experimental thing without using intricate grammar. We can use admonitions also: https://docusaurus.io/docs/next/markdown-features/admonitions

@mjambon I dared to write some educated guesses on how generic mode works (that it does not really understand the syntax of the given language), please check below. This should make it very obvious to reader.

```suggestion
Semgrep can reliably understand the syntax of natively supported languages. The generic mode is useful for [supported languages](/supported-languages/), bringing specific limitations.

:::caution
- Generic mode is not good for detecting malicious code.
- The quality of results in the generic mode can vary depending on the language you use it for.
:::

The generic mode works fine with any human-readable text, as long as it is primarily based on ASCII symbols. Since the generic mode does not understand the syntax of the language you are scanning, the quality of the result may differ from language to language or even depend on specific code. As a consequence, the generic mode works well for some languages, but it does not give consistent results always. Generally, it's possible or even easy to write code in weird ways that prevent generic mode from matching. Note generic mode is not good for detecting malicious code.

**Example**: In XML, one can write `&#x48;&#x65;&#x6C;&#x6C;&#x6F` instead of `Hello`. If a rule pattern in generic mode is `Hello`, Semgrep is unable to match the `&#x48;&#x65;&#x6C;&#x6C;&#x6F`, unlike if it had full XML support.
```

Reply:
@mjambon if my suggestion is wrong or misleading in how generic mode works, feel free to rewrite what I wrote.

Reply:
@adamkvitek it's nice. I'll leave out "Generic mode is not good for detecting malicious code" because it's true of semgrep in general.